plugins/cortex_v2/icon_cortex_v2/actions/bulk_analyze/action.py

```diff
@@ -28,34 +29,42 @@ def run(self, params={}):
         data_type = attributes.get("dataType", None)
         tlp_num = attributes.get("tlp", None)
         # get list of analyzers
-        all_analyzers = api.analyzers.find_all(query="")
+        all_analyzers = api.search_for_all_analyzers()
         # get list of cortex analyzers
         for analyzer in all_analyzers:
-            cortex_analyzers.add(analyzer.json()["name"])
+            cortex_analyzers.add(analyzer.get("name"))
         # check analyzers in list and available
         if not analyze_all:
-            missing_ids = set(analyzer_ids).difference(cortex_analyzers)
-            if len(missing_ids) > 0:
+            missing_ids = set(analyzer_names).difference(cortex_analyzers)
+            if missing_ids:
                 self.logger.error(f"Error Analyzers: {missing_ids} not found in Cortex")
                 # remove missing analyzers
                 for analyzer in missing_ids:
                     self.logger.debug(f"Removing {analyzer}")
-                    del analyzer_ids[analyzer_ids.index(analyzer)]
+                    del analyzer_names[analyzer_names.index(analyzer)]
 
             # loop through analyzers and run
-            for analyzer in analyzer_ids:
-                self.logger.debug(f"Running Analyzer: {analyzer}")
-                job = api.analyzers.run_by_name(
-                    analyzer, {"data": observable, "dataType": data_type, "tlp": tlp_num}, force=1
+            for analyzer_name in analyzer_names:
+                job_results.append(
+                    self.run_analyzer(analyzer_name, {"data": observable, "dataType": data_type, "tlp": tlp_num})
                 )
-                job_results.append(job_to_dict(job, api))
         else:
             # Analyze all
-            for analyzer in cortex_analyzers:
-                job = api.analyzers.run_by_name(
-                    analyzer, {"data": observable, "dataType": data_type, "tlp": tlp_num}, force=1
+            for analyzer_name in cortex_analyzers:
+                job_results.append(
+                    self.run_analyzer(analyzer_name, {"data": observable, "dataType": data_type, "tlp": tlp_num})
                 )
-                job_results.append(job_to_dict(job, api))
-
         # results
         return {Output.JOBS: job_results}
+
+    def run_analyzer(self, analyzer_name, data):
+        self.logger.debug(f"Running Analyzer: {analyzer_name}")
+        analyzer_search = self.connection.API.get_analyzer_by_name(analyzer_name)
+        analyzer_id = analyzer_search.get("id")
+        if not analyzer_id:
+            raise PluginException(f"Analyzer {analyzer_name} not found")
+        job = filter_job(self.connection.API.run_analyzer(analyzer_id, data))
+        if not job or not isinstance(job, dict) or "id" not in job:
+            raise PluginException(f"Failed to receive job from analyzer {analyzer_name}")
+        job["artifacts"] = filter_job_artifacts(self.connection.API.get_job_artifacts(job["id"]))
```

Comment:
Technically, the safety of using `.get()` to avoid `KeyError`s here is unnecessary because line 67 enforces that `"id"` must be in `job` by the time line 69 runs.
However, I will follow your suggestion anyway, because `.get()` is simply good practice :)